go.mod

```diff
@@ -2,6 +2,8 @@ module github.com/aquasecurity/trivy
 
 go 1.18
 
+replace github.com/aquasecurity/defsec => github.com/sm171190/defsec v0.82.7-0.20230112035505-b8f2b199bdce
```

Comment:
note: update when https://github.com/aquasecurity/defsec/pull/1111 is merged

Reply:
I believe the commit you need on defsec/master is `046ee90ace59c3d87dab96889a87b636fdb205ef`

Reply:
thanks , I'll replace it

Reply:
@simar7 please create a new tag for DefSec so trivy can get an [update package](https://github.com/aquasecurity/defsec/pkgs/container/defsec) with new policies ans i'l be able to complete testing

Reply:
We plan to include some other PRs (unrelated to this) in defsec before we can release a new version (and tag bundle) for defsec. We plan to do this before the next release of Trivy so Trivy can use it.

You could pin this to the commit I mentioned and then later when there's a new defsec release available make another PR (to Trivy) to update the version of defsec. That way you aren't blocked.

WDYT?

Reply:
cc @giorod3

Reply:
@simar7 I assume you could release 2 tags one now and one later , these blocking us from completing testing this PR and cis-benchmark PR which is part of next `trivy` release (end of this month)

Reply:
I think it will be resolved if we disable downloading policies.

Reply:
Then, embedded policies will be used. A new OPA bundle is not needed.

Reply:
Maybe the process which produce the policies bundles should be detached from the `defsec` release cycle.  run base on a cron and produce a new bundle only if it find a change , wdyt ?

Reply:
At the moment, @chen-keinan can use the new policies by `trivy k8s --reset && trivy k8s --skip-policy-update` without the new bundle distribution.

>it should run base on a cron and produce a new bundle only if it find a change

We need to sync the OCI artifact version (`ghcr.io/aquasecurity:defsec:0`) with the defsec schema version. If we change the policy schema, the artifact tag should be like `ghcr.io/aquasecurity/defsec:1` so that old Trivy versions keep working. Cron job seems to make it difficult to add breaking changes.

Reply:
> keep working. Cron job seems to make it difficult to add breaking changes.
>
versioning the schema will help introducing breaking changes

Reply:
>for final testing need to make sure it works well with bundle and for users as well

I think the next version will most likely disable policy downloading. @simar7 will decide it, though.

>versioning the schema will help introducing breaking changes

Could you elaborate on that? Do you mean a cron will read the schema version and tag the artifact?

Reply:
> I think the next version will most likely disable policy downloading. @simar7 will decide it, though.
> 

its confusing as currently it download policies by default ?

> Could you elaborate on that? Do you mean a cron will read the schema version and tag the artifact?
>
yes bundle will be tagged with schema version and trivy will know to download the relevant bundle based on what it support

Reply:
No worries - releasing early and often is good anyways. @chen-keinan I've released a new bundle here https://github.com/aquasecurity/defsec/pkgs/container/defsec please take a look.

Reply:
>its confusing as currently it download policies by default ?

The latest version (v0.36.0) uses embedded policies and the main branch downloads policies. I meant we could kind of like revert the change before cutting v0.37.0. We can disable it instead of reverting.
If we confirm there is backward compatibility between the old schema and the new engine and forward compatibility between the new schema and the old engine, we will ship the feature in v0.37.0.